precice.pyx

```diff
@@ -158,41 +480,168 @@ cdef class Interface:
         self.thisptr.setMeshQuadWithEdges (mesh_id, first_vertex_id, second_vertex_id, third_vertex_id, fourth_vertex_id)
 
     # data access
-    # hasData
     def has_data (self, str data_name, mesh_id):
+        """
+        Checks if the data with given name is used by a solver and mesh.
+
+        Parameters:
+            data_name (string): the name of the data
+            mesh_id (int): ID of the associated mesh
+
+        Returns:
+            tag (bool): True if the mesh is used.
+        """
         return self.thisptr.hasData(convert(data_name), mesh_id)
 
     def get_data_id (self, str data_name, mesh_id):
+        """
+        Returns the ID of the data associated with the given name and mesh.
+
+        Parameters:
+            data_name (string): the name of the data
+            mesh_id (int): ID of the associated mesh
+
+        Returns:
+            id (int): ID of the corresponding data
+        """
         return self.thisptr.getDataID (convert(data_name), mesh_id)
 
     def map_read_data_to (self, to_mesh_id):
+        """
+        Computes and maps all read data mapped to the mesh with given ID.
+        This is an explicit request to map read data to the Mesh associated with toMeshID.
+        It also computes the mapping if necessary.
+
+        Parameters:
+            to_mesh_id (int): Id of mesh to map the read data to.
+
+        Previous calls:
+            A mapping to to_mesh_id was configured.
+        """
         self.thisptr.mapReadDataTo (to_mesh_id)
 
     def map_write_data_from (self, from_mesh_id):
+        """
+        Computes and maps all write data mapped from the mesh with given ID. This is an explicit request
+        to map write data from the Mesh associated with fromMeshID. It also computes the mapping if necessary.
+
+        Parameters:
+            from_mesh_id (int): ID from which to map write data
+
+        Previous calls:
+            A mapping from from_mesh_id was configured.
+        """
         self.thisptr.mapWriteDataFrom (from_mesh_id)
 
     def write_block_vector_data (self, data_id, value_indices, values):
+        """
+        Writes vector data given as block. This function writes values of specified vertices to a dataID.
+        Values are provided as a block of continuous memory. valueIndices contains the indices of the vertices
+        The 2D-format of values is (d0x, d0y, d1x, d1y, ..., dnx, dny)
+        The 3D-format of values is (d0x, d0y, d0z, d1x, d1y, d1z, ..., dnx, dny, dnz)
+
+        Parameters:
+            data_id (int): ID to write to.
+            value_indices (array): Indices of the vertices.
+            values (pointer): a pointer to the vector values.
+
+        Previous calls:
+            count of available elements at values matches the configured dimension * size
+            count of available elements at valueIndices matches the given size
+            initialize() has been called
+
+        Refer:
+            Interface::set_mesh_vertex()
```

Comment:
Why do we refer to `set_mesh_vertex()`?

Reply:
I actually do not have a good explanation for this. It was there before so I left it there. I also see no relevance.

Reply:
I think it comes from the doxygen documentation of the c++ interface (see https://github.com/precice/precice/blob/6630c77a76b51503a09cd4c2f8425d976aa46bb9/src/precice/SolverInterface.hpp#L598). The main purpose here is to provide a link in order to be able to quickly navigate. 

If we use `help(...)` for showing the documentation this feature does not work. I am not sure how `readthedocs` deals with this keyword.

For now I would suggest to remove all the `Refer:` sections, since without links they do not really help. As soon as we are using `readthedocs` we might consider reintroducing them.